pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -141,19 +141,19 @@ func (p *Provider) createAMIOptions(ctx context.Context, nodeTemplate *v1alpha1.
 		return nil, err
 	}
 	// Get constrained security groups
-	securityGroupsIDs, err := p.securityGroupProvider.List(ctx, nodeTemplate)
+	securityGroups, err := p.securityGroupProvider.List(ctx, nodeTemplate)
 	if err != nil {
 		return nil, err
 	}
-	if len(securityGroupsIDs) == 0 {
+	if len(securityGroups) == 0 {
 		return nil, fmt.Errorf("no security groups exist given constraints")
 	}
 	return &amifamily.Options{
 		ClusterName:             settings.FromContext(ctx).ClusterName,
 		ClusterEndpoint:         p.ClusterEndpoint,
 		AWSENILimitedPodDensity: settings.FromContext(ctx).EnableENILimitedPodDensity,
 		InstanceProfile:         instanceProfile,
-		SecurityGroupsIDs:       securityGroupsIDs,
+		SecurityGroupsIDs:       p.securityGroupProvider.GetSecurityGroupIds(securityGroups),
```

Comment:
Can we change this to be `SecurityGroups` now that we have the whole struct? I'm not convinced that we need both sets of functions? I think we should generally default to removing functions if one function can do the job of two like is done here.